modules/gapi/test/gapi_planar_test.cpp

```diff
@@ -0,0 +1,211 @@
+        toPlanar(rgb, out);
+    }
+};
+
+struct PlanarTest : public TestWithParam <std::pair<cv::Size, cv::Size>> {};
+TEST_P(PlanarTest, Resize3c3p)
+{
+    cv::Size in_sz, out_sz;
+    std::tie(in_sz, out_sz) = GetParam();
+    int interp = cv::INTER_NEAREST;
+
+    cv::Mat in_mat = cv::Mat(in_sz, CV_8UC3);
+    cv::randn(in_mat, cv::Scalar::all(127.0f), cv::Scalar::all(40.f));
+
+    cv::Mat out_mat     = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+    cv::Mat out_mat_ocv = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+
+    cv::GMat in;
+    auto out = GResize3c3p::on(in, out_sz, interp);
+    cv::GComputation c(cv::GIn(in), cv::GOut(out));
+
+    auto pkg = cv::gapi::kernels<OCVResize3c3p>();
+    c.apply(cv::gin(in_mat), cv::gout(out_mat), cv::compile_args(pkg));
+
+    cv::Mat resized_mat;
+    cv::resize(in_mat, resized_mat, out_sz, 0, 0, interp);
+    toPlanar(resized_mat, out_mat_ocv);
+
+    EXPECT_EQ(0, cv::countNonZero(out_mat != out_mat_ocv));
+}
+
+TEST_P(PlanarTest, Resize3p3p)
+{
+    cv::Size in_sz, out_sz;
+    std::tie(in_sz, out_sz) = GetParam();
+    int interp = cv::INTER_NEAREST;
+
+    cv::Mat in_mat = cv::Mat(cv::Size{in_sz.width, in_sz.height*3}, CV_8UC1);
+    cv::randn(in_mat, cv::Scalar::all(127.0f), cv::Scalar::all(40.f));
+
+    cv::Mat out_mat     = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+    cv::Mat out_mat_ocv = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+
+    cv::GMat in;
+    auto out = GResize3p3p::on(in, out_sz, interp);
+    cv::GComputation c(cv::GIn(in), cv::GOut(out));
+
+    auto pkg = cv::gapi::kernels<OCVResize3p3p>();
+
+    c.compile(cv::descr_of(in_mat).asPlanar(3), cv::compile_args(pkg))
+             (cv::gin(in_mat), cv::gout(out_mat));
+
+    for (int i = 0; i < 3; i++) {
+        const cv::Mat in_mat_roi = in_mat(cv::Rect(0,  i*in_sz.height,  in_sz.width,  in_sz.height));
+        cv::Mat out_mat_roi = out_mat_ocv(cv::Rect(0, i*out_sz.height, out_sz.width, out_sz.height));
+        cv::resize(in_mat_roi, out_mat_roi, out_sz, 0, 0, interp);
+    }
+
+    EXPECT_EQ(0, cv::countNonZero(out_mat != out_mat_ocv));
+}
+
+TEST_P(PlanarTest, Pipeline)
+{
+    cv::Size in_sz, out_sz;
+    std::tie(in_sz, out_sz) = GetParam();
+    int interp = cv::INTER_NEAREST;
+
+    cv::Mat in_mat = cv::Mat(cv::Size{in_sz.width, in_sz.height*3/2}, CV_8UC1);
+    cv::randn(in_mat, cv::Scalar::all(127.0f), cv::Scalar::all(40.f));
+
+    cv::Size uv_sz(in_sz.width / 2, in_sz.height / 2);
+
+    cv::Mat y_mat  = cv::Mat(in_sz, CV_8UC1, in_mat.data);
+    cv::Mat uv_mat = cv::Mat(uv_sz, CV_8UC2, in_mat.data + in_mat.step1() * in_sz.height);
+
+    cv::Mat out_mat     = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+    cv::Mat out_mat_ocv = cv::Mat::zeros(out_sz.height*3, out_sz.width, CV_8UC1);
+
+    cv::GMat inY, inUV;
+    auto out = GResize3p3p::on(GNV12toRGBp::on(inY, inUV), out_sz, interp);
```

Comment:
Should we try to check BGR CSC as well? Is it redundant?

Reply:
This is not a comprehensive testing of preproc-like pipeline with any possible configurations, it just tests that such kind of pipelines are supported by the framework with metadata inferred and kernels launched correctly. I believe that BGR csc is redundant in this test.